lib/resty/openssl/auxiliary/nginx_c.lua

```diff
@@ -115,11 +118,16 @@ end
 get_socket_ssl = function(sock)
   local u = sock[SOCKET_CTX_INDEX]
 
-  local ret
+  local ret, status
   if stream_subsystem then
     ret = C.ngx_stream_lua_resty_openssl_aux_get_socket_ssl(u, void_pp)
   else
-    ret = C.ngx_http_lua_resty_openssl_aux_get_socket_ssl(u, void_pp)
+    status, ret = pcall(function()
```

Comment:
let's not do the try call inside get_socket_ssl function, as it may get performance penalty; instead move it to module level

Reply:
We also need to cover in the test to run the`ngx_http_lua_resty_openssl_aux_get_socket_ssl` path. Please take a look at the `CI_SKIP_NGINX_C` env on how are we previously testing the plain FFI vs. the C API

Reply:
changed